### PR TITLE
Option to update filter list android

### PR DESCRIPTION
**Pull request by deeppandya**
Resolves https://github.com/brave/brave-browser/issues/35270

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Review by tapanmodh — APPROVED** [review]
lgtm

File: browser/ui/android/strings/android_brave_strings.grd [hunk 1]
```diff
@@ -3983,6 +3983,12 @@ If you don't accept this request, VPN will not reconnect and your internet conne
     <message name="IDS_IN_APP_UPDATE_TEXT" desc="Text for in app update snackbar">
       Update is downloaded.
     </message>
+    <message name="IDS_UPDATE_FILTER_LIST_SUCCESS_TEXT" desc="Text to notify users for updating filterlist successfully">
+      Filterlist is updated successfully
+    </message>
+    <message name="IDS_UPDATE_FILTER_LIST_FAIL_TEXT" desc="Text to notify users for the failure of updating filterlist">
+      Filterlist could not be updated. Please try again
+    </message>
```
Comment (mkarolin) [inline browser/ui/android/strings/android_brave_strings.grd:3991]: I think on desktop we use `Filter list` as 2 words.

Comment (deeppandya) [reply browser/ui/android/strings/android_brave_strings.grd:3991]: Updating now. Thank you @mkarolin

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Pixel 6` running `Android 14` using the following build(s):

```
Brave | 1.68.70 Chromium: 126.0.6478.26 (Official Build) canary (64-bit)
--- | ---
Revision | 28f8af9c7d27ab764fc358d8e3c5735d3139d267
OS | Android 14; Build/AP21.240305.006; 34; REL
```

Using the example via https://github.com/brave/brave-core/pull/23929#issue-2327138822, went through the following STR/Cases:

* ensured that there's a new `UPDATE` button available via `Hamburger Menu` -> `Brave Shields & privacy` -> `Content Filtering`
* ensured that tapping `UPDATE` doesn't case any issues/crashes or visible regressions
* ensured that tapping `UPDATE` contacts `https://go-updater.brave.com/extensions` without any issues

**`Example`** | **`Example`**
----------------|-----------------
Screenshot 2024-06-06 at 11 31 22 PM | Screenshot_20240606-233327

**Pull request merged**